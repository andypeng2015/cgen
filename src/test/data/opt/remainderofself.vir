module foo

export function w $foo(w %x) {
@start:
  %c = eq.w %x, 0_w
  br %c, @zero, @notzero
@zero:
  ret %x
@notzero:
  %y = rem.w %x, %x
  ret %y
}
