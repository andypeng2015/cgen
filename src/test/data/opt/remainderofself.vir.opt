module foo

export function w $foo(w %x) {
@2:
  ret 0x0_w
}
