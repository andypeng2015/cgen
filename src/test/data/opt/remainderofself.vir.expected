module foo

export function w $foo(w %x) {
@3:
  ret 0x0_w
}
